openedx/core/djangoapps/credit/tasks.py

```diff
@@ -0,0 +1,51 @@
+""" This file contains celery tasks for credit course views """
+
+from .api import set_credit_requirements
+from celery.task import task
+from celery.utils.log import get_task_logger
+from opaque_keys import InvalidKeyError
+from opaque_keys.edx.keys import CourseKey
+from openedx.core.djangoapps.credit.exceptions import InvalidCreditRequirements
+from xmodule.modulestore.django import modulestore
+from xmodule.modulestore.exceptions import ItemNotFoundError
+
+
+LOGGER = get_task_logger(__name__)
+
+
+@task()
+def update_course_requirements(course_id):
+    """ Updates course requirements table for a course.
+
+     Args:
+        course_id(str): A string representation of course identifier
+
+    Returns:
+        None
+    """
+    try:
+        course_key = CourseKey.from_string(course_id)
```

Comment:
We should probably check if the course is a credit course at this point, rather than waiting for an exception on `set_credit_requirements`.  That way, we avoid loading the course and scanning the course tree for courses that don't have credit enabled.  It's probably worth adding an additional function to the API: something like `is_credit_course(course_key) --> bool`